TombEngine/Objects/TR3/Entity/tr3_civvy.cpp

```diff
@@ -2,418 +2,601 @@
-						item->Animation.TargetState = CIVVY_STATE_IDLE;
+					if (Lara.TargetEntity != &item && AI.ahead)
+						item.Animation.TargetState = CIVVY_STATE_IDLE;
 					else
-						item->Animation.TargetState = CIVVY_STATE_RUN_FORWARD;
+						item.Animation.TargetState = CIVVY_STATE_RUN;
 				}
-				else if (creature->Mood == MoodType::Bored ||
-					(item->AIBits & FOLLOW && (creature->ReachedGoal || laraAI.distance > SQUARE(SECTOR(2)))))
+				else if (creature.Mood == MoodType::Bored ||
+					(item.AIBits & FOLLOW && (creature.ReachedGoal || targetDistance > SQUARE(BLOCK(2)))))
 				{
-					if (item->Animation.RequiredState != NO_STATE)
-						item->Animation.TargetState = item->Animation.RequiredState;
-					else if (AI.ahead)
-						item->Animation.TargetState = CIVVY_STATE_IDLE;
+					if (!creature.Alerted && Random::TestProbability(CIVVY_WAIT_CHANCE))
+						item.Animation.TargetState = CIVVY_STATE_WAIT;
 					else
-						item->Animation.TargetState = CIVVY_STATE_RUN_FORWARD;
+						item.Animation.TargetState = CIVVY_STATE_IDLE;
 				}
-				else if (AI.bite && AI.distance < CIVVY_ATTACK0_RANGE)
-					item->Animation.TargetState = CIVVY_AIM0;
-				else if (AI.bite && AI.distance < CIVVY_ATTACK1_RANGE)
-					item->Animation.TargetState = CIVVY_AIM1;
+				else if (creature.Enemy != nullptr && AI.bite && AI.distance < CIVVY_ATTACK_CLOSE_PUNCH_RANGE)
+					item.Animation.TargetState = CIVVY_STATE_AIM_CLOSE_PUNCH;
+				else if (creature.Enemy != nullptr && AI.bite && AI.distance < CIVVY_ATTACK_FAR_PUNCH_RANGE)
+					item.Animation.TargetState = CIVVY_STATE_AIM_FAR_PUNCH;
 				else if (AI.bite && AI.distance < CIVVY_WALK_RANGE)
-					item->Animation.TargetState = CIVVY_STATE_WALK_FORWARD;
+					item.Animation.TargetState = CIVVY_STATE_WALK;
 				else
-					item->Animation.TargetState = CIVVY_STATE_RUN_FORWARD;
+					item.Animation.TargetState = CIVVY_STATE_RUN;
 
 				break;
 
-			case CIVVY_STATE_WALK_FORWARD:
-				creature->MaxTurn = CIVVY_WALK_TURN_RATE_MAX;
-				extraHeadRot.y = laraAI.angle;
+			case CIVVY_STATE_WALK:
+
+				creature.MaxTurn = CIVVY_WALK_TURN_RATE_MAX;
 
-				if (item->AIBits & PATROL1)
+				if (AI.ahead)
+					jointHeadRot.y = targetAngle;
+
+				if (item.AIBits & PATROL1)
 				{
-					extraHeadRot.y = 0;
-					item->Animation.TargetState = CIVVY_STATE_WALK_FORWARD;
+					jointHeadRot.y = 0;
+					item.Animation.TargetState = CIVVY_STATE_WALK;
 				}
-				else if (creature->Mood == MoodType::Escape)
-					item->Animation.TargetState = CIVVY_STATE_RUN_FORWARD;
-				else if (creature->Mood == MoodType::Bored)
+				else if (item.AIBits & FOLLOW)
+				{
+					if (creature.ReachedGoal)
+						item.Animation.TargetState = CIVVY_STATE_IDLE;
+					else if (targetDistance > CIVVY_WALK_RANGE)
+						item.Animation.TargetState = CIVVY_STATE_RUN;
+					else
+						item.Animation.TargetState = CIVVY_STATE_WALK;
+				}
+				else if (creature.Mood == MoodType::Escape)
+					item.Animation.TargetState = CIVVY_STATE_RUN;
```

Comment:
Please add brackets around this if block for consistency.

Reply:
Done.

I missed the coding convention part that mentioned to add brackets if is an else or else if structure.

So I've update the full civvy.cpp to add the brackets in those structures